api/legacy.go

```diff
@@ -0,0 +1,60 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/ONSdigital/dp-api-clients-go/v2/zebedee"
+	dphandlers "github.com/ONSdigital/dp-net/handlers"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+func LegacyHandler(ctx context.Context, zc ZebedeeClient) http.HandlerFunc {
+	return dphandlers.ControllerHandler(func(w http.ResponseWriter, r *http.Request, lang, collectionID, accessToken string) {
+		handleLegacy(w, r, lang, collectionID, accessToken, zc)
+	})
+}
+
+func handleLegacy(w http.ResponseWriter, req *http.Request, lang, collectionID, accessToken string, zc ZebedeeClient) {
+	ctx := req.Context()
+
+	urlParam := req.URL.Query().Get("url")
+
+	if urlParam == "" {
+		err := errors.New("url parameter not found")
+		log.Error(ctx, "url parameter not found", err)
+		http.Error(w, "URL not found", http.StatusNotFound)
+		return
+	}
+
+	response, err := zc.GetBulletin(ctx, accessToken, collectionID, lang, urlParam)
```

Comment:
Are we happy with the zebedee model? I wonder if we might want to map the zebedee response to the desired target API model here to hide the ugliness and ensure the FE can be kept clean? This is just a suggestion, so no worries if that isn't the direction your team is taking.

Reply:
That is a good point and there hasn't been any talks around the model really. I created this bulletin type coming from zebedee client based purely on the values required for FE rendering. It might need to be revisited at a later point....